crates/sui-framework/sources/governance/staking_pool.move

```diff
@@ -145,13 +144,21 @@ module sui::staking_pool {
         let delegator = tx_context::sender(ctx);
         let principal_withdraw_amount = balance::value(&principal_withdraw);
 
+        // If the pool is deactivated get the epoch at which it was deactivated,
+        // otherwise get the current epoch.
+        let rewards_epoch = option::get_with_default(&pool.deactivation_epoch, tx_context::epoch(ctx));
```

Comment:
Might be more clean to move this logic to `pool_token_exchange_rate_at_epoch`?

Reply:
Happy to move it :)